.github/workflows/checks.yml

```diff
@@ -98,14 +94,91 @@ jobs:
         run: |
           poetry install
 
+      # ====================== CONTRACTS v2_6 ====================== #
+      - name: Cache contracts v2_6
+        id: cache-contracts-v2_6
+        uses: actions/cache@v3
+        with:
+          path: starknet_py/tests/e2e/mock/contracts_compiled_v2_6
+          key: ${{ runner.os }}-contracts-${{ hashFiles('starknet_py/tests/e2e/mock/contracts_v2_6') }}
+    
+      - uses: actions/checkout@v4
+        with:
+          repository: starkware-libs/cairo
+          path: cairo_v2_6
+          ref: 52140f59df98bcd465a7c5f9c3c2f033bc19c2b2
+
+      - name: Create text file
+        run: |
+          echo "${{ github.workspace }}/cairo_v2_6/Cargo.toml" > ${{ github.workspace }}/starknet_py/tests/e2e/manifest-path
+
+      - name: Compile contracts
+        if: steps.cache-contracts-v2_6.outputs.cache-hit != 'true'
+        run: |
+          poetry run poe compile_contracts_v2_6
+
+      # ====================== CONTRACTS v2 ====================== #
```

Comment:
I think it may make sense to switch the v2 scenarios - we will probably stay at 2.5.X version of a compiler, and update 2.6 to newer versions from now on